drivers/storage/portworx/component/podsecuritypolicies.go

```diff
@@ -0,0 +1,126 @@
+package component
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	"github.com/libopenstorage/operator/pkg/constants"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// PSPComponentName name a component for installing podsecuritypolicies.
+	PSPComponentName = "PodSecurityPolicies"
+)
+
+// RegisterPSPComponent registers a component for installing podsecuritypolicies.
+func RegisterPSPComponent() {
+	Register(PSPComponentName, &podsecuritypolicies{})
+}
+
+func init() {
+	RegisterPSPComponent()
+}
+
+type podsecuritypolicies struct {
+	k8sClient client.Client
+}
+
+func (p *podsecuritypolicies) Initialize(k8sClient client.Client, k8sVersion version.Version, scheme *runtime.Scheme, recorder record.EventRecorder) {
+	p.k8sClient = k8sClient
+}
+
+func (p *podsecuritypolicies) IsEnabled(cluster *corev1.StorageCluster) bool {
+	return true
+}
+
+func (p *podsecuritypolicies) Reconcile(cluster *corev1.StorageCluster) error {
+	for _, policy := range portworxPodSecurityPolicies() {
+		err := p.k8sClient.Create(context.TODO(), &policy)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create %s pod security policy: %s", policy.Name, err)
+		}
+	}
+	return nil
+}
+
+func (p *podsecuritypolicies) Delete(cluster *corev1.StorageCluster) error {
+	for _, policy := range portworxPodSecurityPolicies() {
+		err := p.k8sClient.Delete(context.TODO(), &policy)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *podsecuritypolicies) MarkDeleted() {
+}
+
+func portworxPodSecurityPolicies() []policyv1beta1.PodSecurityPolicy {
+	return []policyv1beta1.PodSecurityPolicy{
+		{
+			ObjectMeta: metav1.ObjectMeta{
```

Comment:
We should add owner reference of the cluster to the PSPs. That way if the StorageCluster is force deleted, Kubernetes will remove the PSPs. We do the same for all almost all other components created by the operator.

Reply:
Updated.